validator/src/types.rs

```diff
@@ -20,6 +20,11 @@ impl ValidationError {
     pub fn add_param<T: Serialize>(&mut self, name: Cow<'static, str>, val: &T) {
         self.params.insert(name, to_value(val).unwrap());
     }
+
+    pub fn with_message(mut self, message: &'static str) -> ValidationError {
```

Comment:
Let's take a Cow

Reply:
Sure!
I also added a doc string to explain its usage.